src/frontend/ocamlmerlin/old/old_command.ml

```diff
@@ -99,9 +96,14 @@ let configure (state : buffer) =
           directory = Misc.canonicalize_filename (Filename.dirname path);
         }
     } in
-  let config = Mconfig.load_dotmerlins config
-      ~filenames:(Option.cons (Option.map ~f:Filename.dirname state.path)
-                    (Option.value ~default:[] state.dot_merlins))
+  let config =
+    match state.dot_merlins with
+    | Some (first :: _) -> (* ignore anything but the first one... *)
+      Mconfig.get_external_config first config
+    | None | Some [] ->
+      match state.path with
+      | None -> config
+      | Some p -> Mconfig.get_external_config p config
```

Comment:
@trefis 

I had to change that part (`Mconfig.gec (Filename.dirname p) config` -> `Mconfig.gec p config`) for things to work with the old protocol.

However I am not sure of what is happening here: If `state.path` contains the path to the file, while would we prefer the path to an arbitrary chosen `.merlin` file ?

The new config reader expect the path to a source file, not a dot merlin file. Especially the Dune one which will be able to give file-specific configs.

Reply:
In some cases caller can pass an explicit .merlin file via a command line argument. In this case we should always use the `dot-merlin-reader`.